### PR TITLE
Add settings for syncing only through wi-fi

### PR DESCRIPTION
**Pull request by anna1901**
https://trello.com/c/EAVwEUAw

### CODE REVIEW

File: AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeViewModel.swift [hunk 1]
```diff
@@ -67,6 +67,10 @@ class ChooseSessionTypeViewModel: ObservableObject {
     }
     
     func syncButtonTapped() {
+        guard !userSettings.syncOnlyThroughWifi || networkChecker.isUsingWifi else {
+            alert = InAppAlerts.noWifiNetworkSyncAlert()
+            return
+        }
```
Comment (pawelgil) [inline AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeViewModel.swift:73]: don't we want this here too like in dashboard?:
```Swift
guard networkChecker.connectionAvailable else {
                 alert = InAppAlerts.noNetworkSyncAlert()
                 isRefreshing = false
                 return
             }
```

File: AirCasting/SessionStopping/SyncTriggeringSesionStopperDecorator.swift [hunk 2]
```diff
@@ -2,18 +2,23 @@
...
 import Foundation
+import Resolver
 
 class SyncTriggeringSesionStopperDecorator: SessionStoppable {
     private let stoppable: SessionStoppable
     private let synchronizer: SessionSynchronizer
     
+    @InjectedObject private var userSettings: UserSettings
+    @Injected private var networkChecker: NetworkChecker
+
```
Comment (pawelgil) [inline AirCasting/SessionStopping/SyncTriggeringSesionStopperDecorator.swift:13]: is that addition needed? It's not used anywhere?

File: AirCasting/SessionViews/StandaloneSessionCardView.swift [hunk 3]
```diff
@@ -58,6 +59,10 @@ struct StandaloneSessionCardView: View {
     var finishAndSyncButton: some View {
         Button(Strings.StandaloneSessionCardView.finishAndSyncButtonLabel) {
             if networkChecker.connectionAvailable {
+                guard !userSettings.syncOnlyThroughWifi || networkChecker.isUsingWifi else {
+                    alert = InAppAlerts.noWifiNetworkSyncAlert()
+                    return
+                }
```
Comment (pawelgil) [inline AirCasting/SessionViews/StandaloneSessionCardView.swift:65]: don't we want a no-connection alert too?

btw. I'll investigate possibilities to unify this and not repeat in each view

File: AirCasting/SessionsSynchronization/Controller/Utils/ScheduledSessionSynchronizerProxy.swift [hunk 4]
```diff
@@ -9,6 +9,36 @@ import Combine
 // be the case tho)
 import class UIKit.UIApplication
 import struct UIKit.UIBackgroundTaskIdentifier
+import Resolver
+
+final class WiFiAwareSessionSynchronizerProxy: SessionSynchronizer {
```
Comment (pawelgil) [inline AirCasting/SessionsSynchronization/Controller/Utils/ScheduledSessionSynchronizerProxy.swift:14]: this deserves its own file. Please also take a look how the other proxy is documented and maybe try to do it as well

File: AirCasting/Utils/NetworkMonitoring/NetworkChecker.swift [hunk 5]
```diff
@@ -21,9 +23,12 @@ final class DefaultNetworkChecker: NetworkChecker {
             if path.status == .satisfied {
                 Log.info(Strings.NetworkChecker.satisfiedPathText)
                 self.connectionAvailable = true
+                self.isUsingWifi = path.usesInterfaceType(.wifi)
+                Log.info("CHANGED WIFI STATUS TO: \(self.isUsingWifi)")
```
Comment (pawelgil) [inline AirCasting/Utils/NetworkMonitoring/NetworkChecker.swift:27]: WHY ALL CAPS THO

File: AirCastingTests/Logic/SessionsSynchronization/SyncProxyTests/WifiAwareSessionSynchronizerTests.swift [hunk 6]
```diff
@@ -0,0 +1,82 @@
+// Created by Lunar on 21/07/2022.
+//
+
+import XCTest
+import Combine
+import Resolver
+@testable import AirCasting
+
+class WifiAwareSessionSynchronizerTests: ACTestCase {
```
Comment (pawelgil) [inline AirCastingTests/Logic/SessionsSynchronization/SyncProxyTests/WifiAwareSessionSynchronizerTests.swift:9]: Awesome! 💯💯💯

Very good tests, already on a pro level. Let's schedule a short call to discuss some further possibilities

**Review by jankrzempek — COMMENTED** [review]
👌🏻

File: AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeViewModel.swift (hunk 1 above)
Comment (anna1901) [reply AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeViewModel.swift:73]: It is present below (but as an if)

File: AirCasting/SessionViews/StandaloneSessionCardView.swift (hunk 3 above)
Comment (anna1901) [reply AirCasting/SessionViews/StandaloneSessionCardView.swift:65]: There is an alert for no connection in line 70
